### PR TITLE
chore(ci): bundle 2 — the scripts/ gate lane, three items in sequence (#5097, #5174, #4738)

### LINKED ISSUE #4738: Audit api watchPatterns against full workspace-dependency closure (deploy gap class)
## Problem

The api service (`api`/`api-eu`/`api-apac`) bundles its `workspace:*` deps into its Docker image, but `deploy/api*/railway.json` `watchPatterns` is a hand-maintained list that has drifted from that dependency closure. A change confined to a bundled-but-unwatched package produces `skippedReason: "No changes to watched files"` on the `prod` push and **silently never deploys**.

Concretely hit `packages/mcp/**` while shipping v0.0.63 (#4733/#4734) — fixed in #4737. But `packages/api`'s workspace deps also include, none of which are currently watched by source:

- `@atlas/okf-bundle`
- `@useatlas/chat`
- `@useatlas/twenty`
- `@atlas/oauth-helper` (transitively via `@atlas/mcp`)

(`@useatlas/daytona`/`@useatlas/e2b`/`@useatlas/chat` may be under `plugins/**`, already watched — verify.)

## Acceptance criteria

- Every source dir of a `packages/api` runtime `workspace:*` dependency (transitively, for anything bundled into the api image) is covered by each api-region `watchPatterns`.
- `scripts/check-railway-watch.sh` is extended (or a sibling gate added) to assert the **workspace-dependency closure** is watched — not just explicit Dockerfile `COPY` sources — so this class can't silently recur. The current gate is blind to the broad `COPY . .` that bundles all package source.

## Key files
- `deploy/api/railway.json`, `deploy/api-eu/railway.json`, `deploy/api-apac/railway.json`
- `scripts/check-railway-watch.sh`
- `packages/api/package.json` (dependency closure source of truth)

## Context
Discovered during the v0.0.63 release: all 4 tag-gated services SKIPPED the prod push because the only changes were `packages/mcp/**`. See #4737.

### LINKED ISSUE #5097: chore(test): close the byte-blessing class in the mutation gate — every unmeasured cell, not just dead anchors
Carved out of #5077 when its round-2 `/review-panel` yield ROSE (17 → ~22) and the live-stop rule fired. #5077 landed the verified core; this is what was removed rather than shipped half-done.

## The class, stated once

`--check` compares BYTES. So **any** cell that records "this measured nothing" becomes, once committed, the expected output forever — and the gate then certifies a table that measures nothing. #5077 closed exactly one instance of this (`Cell.anchorFailed` + `anchorFailures()` + a refusal in `mutate.ts`) and its own docstring states the discriminating rule: refuse when the cell *"measured nothing"*, keep it a flag when it is *"a real measurement of a real hang"*.

The other members of the class were not closed.

## 1. `measure()`'s SKIP cell — REMOVED from #5077, needs reinstating properly

A mutated run that skips tests was flagged `⚠️ SKIPPED N — count would be deflated` and written. Reproduced end-to-end by two independent reviewers against a `describe.skipIf(...)` target — the `describeIfPg` shape the guard was written for:

```
[1/1] SKIP  answer returns the wrong number @ subject: 2 test(s) did not run…
wrote scripts/mutations/f.md          rc=0
| answer returns the wrong number | ⚠️ SKIPPED 2 — count would be deflated |
gate --all on that committed table → "all 1 verified table(s) current."  rc=0
```

By the runner's own criterion a skip *measured nothing*, so it belongs on the refuse side. The detection code was correct; only its disposition was wrong. It was removed in #5077 rather than shipped, so **today a mutated run that skips renders a deflated count as an honest number** — strictly worse than the removed version, and the reason this issue is `ready-for-agent` rather than backlog.

`bun printed no pass/fail summary (compile or import error)` is a third member.

## 2. The refusal should be one mechanism

Suggested shape, from the type reviewer:

```ts
readonly notMeasured?: true;              // on Cell, set by every "measured nothing" branch
export function unmeasuredRows(rows): { label: string; reason: string }[]
```

`mutate.ts` refuses on `unmeasuredRows()` instead of `anchorFailures()`. The timeout branch stays UNMARKED — that carve-out is correct and tested.

A stronger version replaces `Cell` with a discriminated union so the tombstone string cannot exist without the state:

```ts
export type Cell =
  | { readonly kind: "count";  readonly fail: number; readonly wholeSuite?: true }
  | { readonly kind: "error";  readonly flag: string }
  | { readonly kind: "anchor"; readonly matches: number };
```

Today `{ kind: "error", fail: 0, flag: "ANCHOR: 0 matches" }` without `anchorFailed` still typechecks, still renders the blessed byte, and `anchorFailures()` returns `[]`. It is constructed in `mutate-core.test.ts` already.

## 3. The two vacuous fixtures, and the harness flaw behind them

`scripts/__tests__/check-mutation-tables.test.sh` shipped five fixtures, each proven sensitive by deleting the guard it pins. Two more were written, **measured vacuous, and removed**: `.todo is REFUSED` and `a DEAD ANCHOR is refused`. Their trees pass `extra='true'` so no table is generated, and `--check` exits 1 with *"is stale"* whether or not the guard exists. Deleting the entire anchor refusal left all seven green.

The root cause is the harness: **`check()` asserts only an exit code**, and exit 1 is shared by STALE, dead anchor, guardrail-4 refusal, unknown-argument and no-specs-found.

- `check()` must also grep the captured output for a per-fixture discriminating phrase.
- The dead-anchor fixture needs a tree carrying a **committed tombstone** — generate it with the refusal bypassed, or hand-write the `.md`. Verified to discriminate: pre-fix runner says `CHECK OK`, post-fix exits 1.
- The `.todo` fixture needs the `.skip` fixture's pre-generation step.

## 4. `--files` misses the transitive-import class

#5077 fixed the named instances (`spec.out`, `../types/` normalisation, `signal-retry.ts`) and **not the class**. Every spec's corpus file is still missing — including `alias-proposal-corpus.ts`, which round 1 named explicitly:

| Spec | Missing dependency |
|---|---|
| `alias-proposal` | `__tests__/alias-proposal-corpus.ts` |
| `object-cmp` | `__tests__/object-cmp-corpus.ts` |
| `identity-corpus`, `subject-cmp`, `tier-guard` | `__tests__/identity-corpus.ts` |
| `bundle-identity` | 6 files incl. `packages/types/src/conversation.ts` |

These are the highest-risk members: they are data-driven inputs (`for (const c of AGREEMENT_CORPUS) test(...)`), so adding one row changes both the published suite size and every kill count. A corpus-only PR gets *"nothing to verify"* locally. CI's `--all` is the backstop, so severity is bounded — but the local gate is blind to exactly the edit shape it advertises.

Fix: `--files` walks relative + `@atlas/api/*` imports one hop (a BFS with a visited set catches every case above). Patching the six named files reproduces the round-1 outcome.

## 5. Smaller, same spirit

- **`ran: number | null` is fail-open.** `null` disables the accounting arm and is indistinguishable from "we failed to parse `Ran N`". No test pins the footer format, so a bun wording change silently reverts the guard to bucket-chasing. Add a verbatim-footer test; consider a notice when a baseline yields `ran === null`.
- **`BaselineProblem.kind` has no exhaustiveness check** at its only consumer (`mutate.ts`'s `deflation` boolean). A sixth kind compiles and silently takes the no-hint branch. Consider `remedy: "check-pg" | "fix-the-tree" | "check-the-path"` on the problem itself.
- **The gate's remedy footer flattens five states into "regenerate"** — wrong for four of them (dead anchor, RED baseline, unloadable spec, missing target). Split `mutate.ts`'s exit codes: 2 = unmeasurable, 1 = stale.
- **`render` is not fully deterministic**: timeout cells bake `round(timeoutMs/1000)` where the timeout derives from baseline duration, so any `-pg` timeout cell is machine-dependent — and `--check` is now a required CI gate. Render `⚠️ timed out`, keep seconds on the console.
- **The generated legend documents only `ANCHOR`**, which post-fix can no longer reach a committed table, and explains none of the flags that can.
- **No mutation rows for `baselineProblem` / `anchorFailures`** in `mutation-core.mutations.ts` — the one module whose table exists to price its own tests. Manual sweep scored 14/14, so this is a completeness gap, not a hidden weakness.
- **`--affected`'s positive selection is uncovered** — the `realpath` normalisation, the `spec.out` hand-edit path, and untracked-file selection. Finding #4's rename bug is exactly what that hole let through.

## Acceptance criteria

- [ ] One refusal mechanism covers every "measured nothing" cell — anchor, skip/todo under mutation, and no-summary — with the timeout carve-out intact and documented
- [ ] `measure()` detects a deflated mutated run again, and the run REFUSES rather than committing the cell
- [ ] `check()` asserts a discriminating phrase, not just an exit code
- [ ] The `.todo` and dead-anchor fixtures are reinstated and each proven sensitive by deleting the guard it pins
- [ ] `--files` walks imports; a corpus-only edit selects its spec
- [ ] A verbatim `Ran N tests across M files` footer is pinned by a test

## Falsification

- [ ] For each refusal: build the committed-tombstone tree, confirm the pre-fix runner says `CHECK OK` and the post-fix one exits non-zero
- [ ] Delete each guard in turn and confirm exactly the fixture that pins it goes red — the measurement that found the two vacuous ones
- [ ] A corpus-only diff selects its spec under `--affected`

## Dependencies

Depends on #5077. Nothing else.

### LINKED ISSUE #5174: Lighthouse comment renderer is untested and its report path is duplicated six ways
The Lighthouse PR-comment renderer is ~250 lines of JavaScript embedded in a YAML `script:` block, and the report path it reads is duplicated across six sites. Neither is reachable by any gate in this repo: oxlint does not parse YAML, `bun run type` never sees it, `scripts/test-others.ts` only discovers workspace packages, and there is no `actionlint`/`yamllint` step anywhere in `ci.yml`. Remote CI validates `lighthouse.yml` only by executing it.

This came out of #4899's review. That PR fixed the reports pipeline and hardened the comment, but its verification lives in a throwaway out-of-tree harness — 138 assertions and 48 killed mutations that no one can re-run. Two reviewers independently flagged that the class-closure it claims is guarded only by review attention, which is precisely what failed: six separate false-or-overreaching message claims were caught across two review rounds, four of them introduced by the fix for a previous one.

It was deliberately deferred rather than built inside a stop round, because it is new machinery entering a diff with no review round left to audit it.

## Why it is worth doing

- **The renderer is invisible to every gate.** A syntax error in that script block is caught by running the workflow, on a PR that happens to touch the filtered paths.
- **The report path is duplicated six ways** — `lighthouserc.js`'s `upload.outputDir`, the verify step's `dir=`, the artifact `path:`, the two comment-script directory literals, and `.gitignore`. #4899 was born of exactly this shape: a `mv` target that no longer agreed with what lhci wrote. The `lighthouserc.js` → workflow direction now fails loudly (the verify step exits non-zero when nothing lands), and `readManifest` warns on a missing manifest, so the remaining silent class is narrow — but it is still silent.
- **The wording ratchet is lexical and out-of-tree.** The retired wordings that must never reappear in a rendered comment are pinned only in the uncommitted harness.

## Sketch

1. Extract the script to `.github/scripts/lighthouse-comment.js` exporting `buildBody({ fs, path, core, env })`, and have the workflow `require` it. This kills the drift structurally rather than by convention, and brings the file under oxlint.
2. Commit the four probes as a test the repo already knows how to run. There are 16 `scripts/__tests__/*.test.sh` fixtures auto-globbed by `scripts/ci-local.sh`, but note CI registers each one explicitly in `ci.yml`'s `drift` job — so this is ~4 registrations, not 1.
3. A `scripts/check-lighthouse-report-paths.sh` that derives `lighthouse-reports/<form-factor>` from `lighthouserc.js` and asserts the workflow's literals agree, plus its own adversarial fixture.
4. Consider a repo-wide `actionlint` step while the wiring is open — nothing validates any workflow today.

## Key files

- `.github/workflows/lighthouse.yml` — the inline renderer and the duplicated literals
- `lighthouserc.js` — `upload.outputDir`, the source of truth for the path
- `scripts/__tests__/` + `.github/workflows/ci.yml` (`drift` job) — the existing lane and its registration pattern
- `scripts/ci-local.sh` — `g_gate_fixtures` auto-glob

## Acceptance criteria

- The comment renderer lives in a committed file that the workflow requires, and is covered by a test CI runs
- A guard fails when `lighthouserc.js`'s `outputDir` and the workflow's report-path literals disagree, with an adversarial fixture proving the guard can fail
- The retired-wording ratchet is enforced in CI rather than in a scratch harness
- Reference: the out-of-tree harness from #4899's review had 138 assertions and 48 mutations; the committed version does not need all of it, but the empty-state outcome words and the escalation gate are the two worth keeping

## Dependencies

Depends on the #4899 fix being merged first (PR #5170) — this refactors the file that PR rewrites.

### LINKED ISSUE #5296: chore(ci): bundle 2 — the scripts/ gate lane, three items fixed in sequence (#5097, #5174, #4738)
Three items on one surface — the `scripts/` gate lane and its `ci.yml` `drift`-job registrations — bundled so **one** review panel reads the combined diff instead of three panels reading three. **One commit each, in the order below**, so a bad one can be dropped without losing the others (the #5289 / #5290 / #5247 pattern).

Bundling these three rather than shipping them in parallel also dissolves a real collision: #5174 and #5097 both register fixtures in `ci.yml`'s `drift` job, so as separate PRs one of them rebases.

## Order, and why it is load-bearing

1. **#5097 first — the mutation gate currently certifies a table that measures nothing.** `--check` compares BYTES, so a cell recording "this measured nothing" becomes the expected output forever. A mutated run that SKIPS tests renders a deflated count as an honest number today. It is the only live defect of the three; the other two are missing coverage.
2. **#5174 second** — it establishes the `.github/scripts/` + fixture-registration pattern (~4 registrations in `ci.yml`, not 1). Landing it after #5097 means the registration edits accumulate in one file in a known order.
3. **#4738 third** — extends `scripts/check-railway-watch.sh` to assert the workspace-dependency closure. Independent of the other two except for the shared lane.

## Members

- **#5097** — close the byte-blessing class in the mutation gate: every unmeasured cell, not just dead anchors. Reinstate `measure()`'s SKIP disposition on the refuse side, and make the refusal ONE mechanism (`notMeasured` + `unmeasuredRows()`), keeping the timeout carve-out unmarked.
- **#5174** — extract the ~250-line Lighthouse comment renderer out of the YAML `script:` block into a committed, oxlint-visible file the workflow requires; guard the 6-way duplicated report path against `lighthouserc.js`'s `outputDir`; commit the retired-wording ratchet that lives only in an out-of-tree harness.
- **#4738** — every source dir of a `packages/api` runtime `workspace:*` dependency is covered by each api-region `watchPatterns`, with a gate asserting the closure so the silent-no-deploy class cannot recur.

## Key files

- `scripts/mutate.ts`, `scripts/mutations/` — #5097
- `.github/workflows/lighthouse.yml`, `lighthouserc.js`, `.github/scripts/`, `scripts/__tests__/` — #5174
- `deploy/api/railway.json`, `deploy/api-eu/railway.json`, `deploy/api-apac/railway.json`, `scripts/check-railway-watch.sh` — #4738
- `.github/workflows/ci.yml` (`drift` job) — **all three**, which is why they are one PR

## Acceptance criteria

- [ ] Each member's own acceptance criteria are met in full — read each issue, do not work from this summary
- [ ] One commit per member, in the order above, each commit message naming its issue
- [ ] Every new guard has an adversarial fixture proving it CAN fail — a guard with no measured RED is the exact class #5097 exists to close, and shipping one here would be self-refuting
- [ ] `ci.yml` registrations for all three land coherently, with no member's fixture unregistered
- [ ] If any member has to be dropped, say which and why — do not silently narrow the bundle

## Dependencies

None open. #5170 (the #4899 fix #5174 refactors on top of) is merged and closed.

### PR DESCRIPTION
**Pull request by msywulak**
Bundle 2 — the `scripts/` gate lane. Three items on one surface, **one commit per member in the stated order**, so a bad one can be dropped without losing the others.

Closes #5296
Closes #5097
Closes #5174
Closes #4738

| # | Commit | What it closes |
|---|---|---|
| 1 | `test(mutate)` | the byte-blessing class in the mutation gate — **the only live defect of the three** |
| 2 | `refactor(ci)` | the Lighthouse renderer, its retired-wording ratchet, and a report-path closure gate |
| 3 | `fix(deploy)` | the silent-no-deploy class: bundled workspace paths nothing watches |

All three touch `ci.yml`'s `drift` job, which is why they are one PR. The registrations accumulated in one file in a known order and the fixture-count arithmetic there is correct at the end: **23 suites, 22 named in `drift`** plus `scan-image.test.sh` in `image-scan.yml`.

---

## ⚠️ Every new guard has an adversarial fixture MEASURED RED

This PR's first member exists to close the class of guards that measure nothing, so shipping an unfalsifiable guard here would be self-refuting. Nineteen falsifiers were built and run — each one deletes the guard a fixture pins and confirms *that* fixture reds while the rest stay green.

### #5097 — three fixtures, three falsifiers

| Falsifier | Result |
|---|---|
| delete `deflationProblem`'s skip/todo arm | fixtures 2 and 2b red — **22/2** |
| delete `mutate.ts`'s `unmeasuredRows` refusal | fixture 6b red — **23/1** |
| revert `--files` to the pre-fix seeds-only list | fixture 6c red — **23/1** |

⚠️ **Falsifier B is the sharpest result in the PR: the exit code stayed 1 and only the PHRASE went missing.** Without the phrase assertion, fixture 6b would have reported `ok` against a runner carrying no refusal at all — which is exactly how #5077 measured it vacuous and deleted it. That is the whole argument for `check()` requiring a discriminating phrase.

The mutation table for the runner's own module is regenerated with six new rows pricing the new guards, all non-zero. `⚠️ unmeasuredRows refuses only the DEAD-ANCHOR member again (#5077's shape)` is the row this change exists for; it kills 1.

One row is **DELETED rather than re-anchored** — *"an errored cell renders as its `fail` number instead of a warning"*. It deleted `renderCell`'s error arm so the `fail: 0` on an error cell would publish as a `0`; that `fail` no longer exists, because `Cell` is now a discriminated union whose no-count variants carry no count to leak. There is nothing to move the anchor to, and a successor pointing at an unreachable state would record a tombstone of its own. The rows that replaced it price the guards that made it unreachable.

### #5174 — 51 assertions, seven falsifiers

| Falsifier | Result |
|---|---|
| reintroduce `"superseded"` in the `cancelled` state | 2 red |
| reintroduce `"wrote no manifest.json"` in `success` | 3 red |
| collapse `skipped` into `"No reports found."` | 1 red |
| name `"a budget assertion failed"` in the banner | 2 red |
| require the renderer by a RELATIVE path | 1 red |
| drop `.github/scripts/` from the lint path list | 1 red |
| **swap the LCP and CLS audit ids** | 2 red |

⚠️ **The last one is why every number in the fixture is different.** With `LCP == CLS`, or with equal category scores, swapping two audit ids or two summary keys renders plausibly and passes — the accidental-equality class. `1234.6` / `0.0721` travel through different formatters to `1235 ms` and `0.07`.

⚠️ **The ratchet runs over RENDERED BODIES, never over the source, and that is load-bearing rather than convenient.** The renderer's comments NAME each retired wording in order to explain why it is retired, so a lexical source scan would flag its own documentation, and the only ways out are deleting the explanation or exempting the file. A guard must parse as the READER parses, and the reader reads the comment. Every ratchet case carries a positive control, because a ratchet alone is satisfied by a renderer that returns the empty string.

### #4738 — 16 cases, five falsifiers

| Falsifier | Result |
|---|---|
| DELETE the whole closure arm | **10 of 16 red** |
| probe the BARE DIRECTORY instead of a source file | 1 red (case 4 exactly) |
| read only `dependencies` (drop peer/optional) | 2 red (the peer member) |
| stop following the graph (roots only) | 3 red |
| revert `deploy/api`'s two new entries | 2 red |

Plus two **false-positive controls**, because a gate that cries wolf gets its errors pasted into `watchPatterns` wholesale: an unbundled workspace package is not demanded, and a package contributing only a data file is not demanded whole.

---

## What each member found

**#5097** — `--check` compares BYTES, so any cell recording *"this measured nothing"* becomes the expected output forever. #5077 closed one member of that class and **removed** the detection for the rest, so today a mutated run that skips tests renders a deflated count as an honest number with no warning anywhere. `Cell` becomes a discriminated union; `unmeasuredRows()` is the single refusal, keyed on the discriminant rather than on any cause, so a fifth member is refused by construction. The timeout carve-out is structural (`runTarget` returns `timedOut: true`) rather than a substring match over prose. `--files` now walks imports one hop, which is what makes a corpus-only edit select its spec — every spec's corpus file was missing from its own dependency list.

**#5174** — the renderer was 250 lines of JS in a YAML `script:` block, reachable by no gate in the repo. It renders a CLAIM about a run, and #4899 was a false one that stood for 1200+ runs. Its report path was duplicated **eight** ways, not the six the issue lists — the two extras are the verify step's diagnostic `find` and a customer-facing claim in `apps/docs`. The new gate EXECUTES `lighthouserc.js` rather than grepping it, and derives the form-factor allowlist from that config's own validation error, because `Object.keys(PROFILES)` is not exported and writing `desktop mobile` in the gate would have been a ninth copy of the thing it exists to stop.

⚠️ **The issue's sketch was wrong about oxlint**: `.github/` is in neither lint path list, so extracting there would have satisfied the letter of the refactor and left the acceptance criterion unmet. Both `lint` and `lint:type-aware` now include `.github/scripts/`, and the fixture asserts that wiring rather than assuming it.

**#4738** — the gate could not fail for the class it was filed for, and the reason is structural: it SKIPS `COPY . .`, and that one line bundles every package in the repo. Before this change it reported *"All 54 COPY sources covered"* and exit 0 while `packages/okf-bundle` was copied into the api runtime image **by name** and watched by nobody — `deploy/docs` watched it; the api, which imports `@atlas/okf-bundle/wire` at runtime, did not.

Two arms now, both derived rather than enumerated. Membership is **by NAME, not by the `workspace:*` protocol** — every plugin declares `@useatlas/plugin-sdk` as a peerDependency RANGE while the image copies it in, and bun links by name ignoring range satisfaction. Coverage is tested with a **synthetic probe file**, not the bare directory path, because `packages/**/package.json` matches the manifest and nothing else and a bare `packages/foo` entry matches the directory while watching none of its source — both read as coverage against a directory test.

Found and fixed: `packages/okf-bundle` and `packages/sdk` in all three api regions, and `packages/sdk` in `web`. **`web` was outside the issue's scope** but the arm is derived rather than enumerated, so it reads that service too and found the same class there; scoping the arm to the api regions to avoid it would have been the enumeration this gate replaces.

---

## Scope decisions, stated rather than silent

- **Nothing was dropped.** All three members shipped in full.
- **`actionlint` is deliberately NOT added.** It is #5174's *"consider"* item, it is repo-wide new machinery, and it belongs in a diff that has a review round left to audit it.
- **#5097's "smaller, same spirit" items** are not in its acceptance criteria and are not here: the machine-dependent seconds in a `-pg` timeout cell, splitting `mutate.ts`'s exit codes 2-vs-1, the generated legend, `BaselineProblem.kind` exhaustiveness, and `--affected`'s positive-selection coverage. The six acceptance criteria and the falsification list are met in full.
- **Counts that drifted as a consequence** were re-measured, not estimated: `ci.yml`'s fixture arithmetic (20→23), `ci-local.sh`'s gate count (36→37, one new `launch`) in CLAUDE.md, `docs/agents/workflow.md`, `.claude/commands/ci.md` and the script's own header, and its Stage-1 launch count (32→33).

## Verification

Pre-flight green on the head SHA: `test-isolated.ts --affected`, `bun run lint`, `bun run type`, `bun run lint:type-aware`.

⚠️ **The local full mutation sweep was deliberately NOT run.** The only Postgres on this machine belongs to a parallel agent worktree, and a mutation table measured on a contended database comes back STALE when nothing drifted — a measurement remote CI then disagrees with. Instead the byte-stability of `render()` was established structurally and confirmed on the one table that had to be regenerated: `mutation-core.md`'s diff changes only the new rows and the suite size, and its committed timeout cell renders identically. `mutation-tables` on this PR is the gate.

---

## ⚠️ Review curve — the panel stopped at the 3-round cap, it did not come back CLEAN

`/ship-issue` requires this disclosure when the loop leaves on the cap rather than
on a clean round. **It did not converge to CLEAN. Read the diff knowing that.**

| Round | Findings | New surface | Defect-in-prior-fix |
|---|---|---|---|
| 1 — three code reviewers | ~39 | 39 | 0 |
| `fix-vs-finding` ×2 | 2 **REPRODUCED** | — | 2 |
| 2 — three code reviewers | ~27 | ~18 | ~9 |
| Closing — `comment-analyzer` | 15 | 15 | — |

The count fell (39 → 27) and new surface dominated both rounds, so neither the
yield rule nor the ratio rule fired. What ended it is the round cap. Every
confirmed must-fix is fixed; the follow-ups below are recorded, not dropped.

### What the extra rounds actually bought

Both `fix-vs-finding` passes came back **REPRODUCED** — my own round-1 fixes each
reproducing the defect they fixed, one layer over, with the closing spelling
already a few lines away in code the same commit had written:

1. `pgHint: boolean` made `{ kind: "empty", pgHint: true }` legal — the exact state
   the field was added to prevent, while the sibling arm pinned `pgHint: false` as a
   literal four lines below.
2. The COPY-source vacuity floor *counted* the vacuity and still printed
   `All 0 COPY sources covered` — the string its own comment quotes as the defect —
   while every other floor in that loop `continue`s.

Neither was found by self-review. Both were found by fresh context.

The closing comment sweep then found **the same class in the guard this PR added
last**: `check-gate-fixtures-wired.sh` claimed its grep distinguished a `run:` line
from a comment. It did not, and the gap was live — two workflow comments carry full
repo-relative paths, so deleting either suite's `run:` step left the gate green off
the prose alone. Measured on a copy of the real tree, then closed. That is the
**sixth** false-guarantee comment review caught here, and the first that was also a
live hole.

⚠️ On a change whose entire subject is guards that make false claims about what
they measured, that count is the most useful thing in this PR. Six of my own
comments asserted guarantees the code did not deliver. None was caught by me.

### Falsifier discipline

~40 falsifiers built and measured RED across the rounds, each deleting the guard one
fixture pins and confirming *that* fixture reds while the rest stay green.

⚠️ **The single most repeated measurement: in every shell falsifier the exit code
stayed unchanged and only the output assertion discriminated.** Five independent
confirmations. That is the whole argument for `check()`'s discriminating-phrase
requirement, and it is why three suites gained more fixtures than the findings
literally asked for.

Two guards initially had NO falsifier and are why some suites grew further: the
per-mutation refusal (the guard of record for #5097's headline claim — deleting it
left every suite in the repo green) and the image-content completeness floor. One
arm remains genuinely unfalsifiable and says so in the code with its measurement.

### Follow-ups — recorded, not dropped

- `runTarget`'s `{ error?, timedOut? }` is the two-fields-that-can-disagree shape
  `Cell` was fixed for, one level up. Unreachable today; wants the `TargetRun` union.
- The generated `## ⚠️ Flagged` legend explains `ANCHOR`, which post-fix can no
  longer reach a committed table, and omits `HANGS`, which can. The text is in
  `render`, so fixing it regenerates both tables carrying a Flagged section — one is
  `warehouse-producer`, a `-pg` spec I cannot regenerate against a contended
  Postgres without producing numbers CI disagrees with. **Filed rather than
  half-fixed.**
- Deriving `wholeSuite` instead of storing it (deletes `countCell` and both wrong
  states); the `ApiRelPath` brand; moving `statCandidate` into `mutation-core.ts` so
  the widen-don't-narrow rule gets a falsifier at all; `pgHint` as a remediation
  enum; the "~16 min" figure quoted at four sites by a header that forbids it; and
  `.claude/rules/testing.md`'s "~27 gates" (it is 38).
- `actionlint`, and #5097's non-AC "smaller, same spirit" items.

### Commit shape

Two commits per member, then the cross-member review commits — so #5296's
one-droppable-unit-per-member property survives: dropping a member means dropping
its pair.

### CODE REVIEW

File: packages/api/scripts/mutate.ts [hunk 1]
```diff
@@ -51,21 +51,24 @@
  * not a triumph of the tests.
  */
 
-import { mkdirSync, writeFileSync } from "node:fs";
+import { existsSync, mkdirSync, readFileSync, statSync, writeFileSync } from "node:fs";
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/scripts/mutate.ts]: Confirmed and fixed in `ab31af9`.

`existsSync` was left behind by this PR's own refactor: `--files` used to guard with `existsSync(abs)` before reading a seed, and `isFile` used `existsSync` + `statSync`. Both were replaced during review — the seed read became a `try`/`catch` (an `existsSync` check is true for a DIRECTORY, so a `target.file` that became one threw `EISDIR` and took the whole gate down with status 1, which this script's caller reads as STALE), and `isFile` became `statCandidate`, using a single `statSync(abs, { throwIfNoEntry: false })` to avoid two syscalls and a TOCTOU window.

So after those two changes the only remaining occurrences were the import and three comments — two of which say the code deliberately uses `statSync` *instead of* `existsSync`. The implementation moved, the comments were updated to document why, and the import was not removed. Those three comments are accurate and are staying; they are the record that the switch was intentional.

Worth noting for anyone reading this thread later: **no gate in this repo was going to catch it.** oxlint's unused-import rule sits in the permanent-`warn` set per ADR-0031, so `bun run lint` exits 0 with it. This finding came from the bot alone.
